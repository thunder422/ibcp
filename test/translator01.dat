##########################################
#  Translator Test #1: Assignment Tests  #
##########################################

A=3
A,B=3
A=B=3
A=(B)=3
A=(B=3)
A,B=(C=3)
A,B=C=4
A(B,C)=D
A(B,C),E=D
E,A(B,C)=D
E,A(B,C),F=D
A(B,(C))=D
A(B+C)=D
A(B=C)=D
A(B+C,D=E)=F
A(B,C)=D(E)
A(B,C),D(E)=INT(F)+Function(G)+Array(H)

# begin of error tests
A=B,C=4
A=B+C,4
A(B+C,(D,E)=F
A(B+C,(D=E)=(F,G)
A,B+C
3=A
A,3,B=4
(A=B)
A,B,(C)=4
A+B
