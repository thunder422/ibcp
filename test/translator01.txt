
Input: A=3
Output: A<ref> 3 Assign 

Input: A,B=3
Output: A<ref> B<ref> 3 AssignList 

Input: A=B=3
Output: A<ref> B 3 = CvtDbl Assign 

Input: A=(B)=3
Output: A<ref> B')' 3 = CvtDbl Assign 

Input: A=(B=3)
Output: A<ref> B 3 =')' CvtDbl Assign 

Input: A,B=(C=3)
Output: A<ref> B<ref> C 3 =')' CvtDbl AssignList 

Input: A,B=C=4
Output: A<ref> B<ref> C 4 = CvtDbl AssignList 

Input: A(B,C)=D
Output: B CvtInt C CvtInt A(<ref> D Assign 

Input: A(B,C),E=D
Output: B CvtInt C CvtInt A(<ref> E<ref> D AssignList 

Input: E,A(B,C)=D
Output: E<ref> B CvtInt C CvtInt A(<ref> D AssignList 

Input: E,A(B,C),F=D
Output: E<ref> B CvtInt C CvtInt A(<ref> F<ref> D AssignList 

Input: A(B,(C))=D
Output: B CvtInt C')' CvtInt A(<ref> D Assign 

Input: A(B+C)=D
Output: B C + CvtInt A(<ref> D Assign 

Input: A(B=C)=D
Output: B C = A(<ref> D Assign 

Input: A(B+C,D=E)=F
Output: B C + CvtInt D E = A(<ref> F Assign 

Input: A(B,C)=D(E)
Output: B CvtInt C CvtInt A(<ref> E CvtInt D( Assign 

Input: A(B,C),D(E)=INT(F)+Function(G)+Array(H)
Output: B CvtInt C CvtInt A(<ref> E CvtInt D(<ref> F INT( G<ref> Function([10:G<ref>] + H CvtInt Array( + AssignList 

Input: A=B,C=4
          ^-- expected operator or end-of-statement

Input: A=B+C,4
            ^-- expected operator or end-of-statement

Input: A(B+C,(D,E)=F
               ^-- expected operator or closing parentheses

Input: A(B+C,(D=E)=(F,G)
                     ^-- expected operator or closing parentheses

Input: A,B+C
          ^-- expected equal or comma for assignment

Input: 3=A
       ^-- expected item for assignment

Input: A,3,B=4
         ^-- expected double variable

Input: (A=B)
       ^-- expected command

Input: A,B,(C)=4
           ^-- expected double variable

Input: A+B
        ^-- expected equal or comma for assignment

