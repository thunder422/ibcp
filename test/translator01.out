
Input: A=3
Output: A = 3 

Input: A,B=3
Output: A, B = 3 

Input: A=B=3
Output: A = B = 3 

Input: A=(B)=3
Output: A = (B) = 3 

Input: A=(B=3)
Output: A = (B = 3) 

Input: A,B=(C=3)
Output: A, B = (C = 3) 

Input: A,B=C=4
Output: A, B = C = 4 

Input: A(B,C)=D
Output: A(B, C) = D 

Input: A(B,C),E=D
Output: A(B, C), E = D 

Input: E,A(B,C)=D
Output: E, A(B, C) = D 

Input: E,A(B,C),F=D
Output: E, A(B, C), F = D 

Input: A(B,(C))=D
Output: A(B, (C)) = D 

Input: A(B+C)=D
Output: A(B + C) = D 

Input: A(B=C)=D
Output: A(B = C) = D 

Input: A(B+C,D=E)=F
Output: A(B + C, D = E) = F 

Input: A(B,C)=D(E)
Output: A(B, C) = D(E) 

Input: A(B,C),D(E)=INT(F)+Function(G)+Array(H)
Output: A(B, C), D(E) = INT(F) + Function(G) + Array(H) 

Input: A=B,C=4
          ^-- expected operator or end-of-statement

Input: A=B+C,4
            ^-- expected operator or end-of-statement

Input: A(B+C,(D,E)=F
               ^-- expected operator or closing parentheses

Input: A(B+C,(D=E)=(F,G)
                     ^-- expected operator or closing parentheses

Input: A,B+C
          ^-- expected equal or comma for assignment

Input: 3=A
       ^-- expected command or item for assignment

Input: A,3,B=4
         ^-- expected double variable

Input: (A=B)
       ^-- expected command or item for assignment

Input: A,B,(C)=4
           ^-- expected double variable

Input: A+B
        ^-- expected equal or comma for assignment

